### PR TITLE
[#IC-115] Create resources for `Selfcare IO`

### PR DESCRIPTION
**Pull request by balanza**
Co-authored-by: Diego Lagos <diegolagospagopa@users.noreply.github.com>

### List of changes

* create a new storage account to contain built frontend artifact
* expose frontend artifacts through a CDN
* create an app service instance to contain backend node application

### Motivation and context

Will create resources in order to have an instance of `SelfCare IO` portal. DNS records and Api Gateway/Api Management set up is out of the scope for this PR
inf

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/core/env/prod/terraform.tfvars [hunk 1]
```diff
@@ -20,15 +20,16 @@ common_rg = "io-p-rg-common"
...
+cidr_subnet_eventhub            = ["10.0.10.0/24"]
+cidr_subnet_fnelt               = ["10.0.11.0/24"]
+cidr_subnet_fnpblevtdispatcher  = ["10.0.12.0/24"]
+cidr_subnet_appgateway          = ["10.0.13.0/24"]
+cidr_subnet_redis_apim          = ["10.0.14.0/24"]
+cidr_subnet_apim                = ["10.0.101.0/24"]
+cidr_subnet_vpn                 = ["10.0.133.0/24"]
+cidr_subnet_azdoa               = ["10.0.250.0/24"]
+cidr_subnet_dnsforwarder        = ["10.0.252.8/29"]
+cidr_subnet_selfcare_fe_storage = ["10.0.137.0/24"]
```
Comment (pasqualedevita) [inline src/core/env/prod/terraform.tfvars:32]: front end storage not need subnet

File: src/core/selfcare_be.tf [hunk 2]
```diff
@@ -0,0 +1,153 @@
+module "selfcare_be" {
+  source = "git::https://github.com/pagopa/azurerm.git//app_service?ref=v2.0.2"
+
+  name = "${var.location}-${var.env_short}"
+
+  ftps_state = "AllAllowed"
```
Comment (pasqualedevita) [inline src/core/selfcare_be.tf:6]:
```suggestion
```
????

File: src/core/selfcare_be.tf [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+  plan_name     = "${var.location}-plan"
+  plan_sku_size = var.app_service_plan_sku
+  plan_sku_tier = var.app_service_plan_sku_tier
+  plan_kind     = var.app_service_plan_kind
```
Comment (pasqualedevita) [inline src/core/selfcare_be.tf:11]:
```suggestion
  plan_kind     = "Linux"
  plan_reserved = true
```
use only linux plans for cost savings.
when we use linux we need also to set

`plan_reserved = true # Mandatory for Linux plan`

ref. https://github.com/pagopa/pagopa-infra/blob/main/src/mock_ec.tf#L28

File: src/core/selfcare_be.tf [hunk 4]
```diff
@@ -0,0 +1,153 @@
+module "selfcare_be" {
+  source = "git::https://github.com/pagopa/azurerm.git//app_service?ref=v2.0.2"
+
+  name = "${var.location}-${var.env_short}"
```
Comment (pasqualedevita) [inline src/core/selfcare_be.tf:4]:
```suggestion
  name = format("%s-app-devportal-be", local.project)
```
missing app service name
we don't use location but we can introduce it. Better to add a `var.location_short`

File: src/core/selfcare_be.tf (hunk 3 above)
Comment (pasqualedevita) [inline src/core/selfcare_be.tf:11]: better to use specific var names, for example

devportal_plan_sku
devportal_plan_sku_tier
devportal_plan_kind

File: src/core/selfcare_be.tf [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  plan_reserved = var.app_service_plan_reserved
+
+  resource_group_name = data.azurerm_resource_group.rg.name
+
+  # Linux App Framework and version for the App Service.
+  linux_fx_version = "${var.runtime_name}|${var.runtime_version}"
+
+  app_settings = {
+    WEBSITE_NODE_DEFAULT_VERSION = "6.11.2"
+    WEBSITE_NPM_DEFAULT_VERSION  = "6.1.0"
```
Comment (pasqualedevita) [inline src/core/selfcare_be.tf:21]: node 6?

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/core/selfcare_be.tf (hunk 5 above)
Comment (balanza) [reply src/core/selfcare_be.tf:21]: Same as https://github.com/pagopa/io-infrastructure-live-new/blob/25bbedf15cb03fa78a10d910303f98337df8522f/prod/westeurope/internal/developerportalbackend/app_service/terragrunt.hcl#L60

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**